Homeworks/05/src/App.tsx

```diff
@@ -1,5 +1,135 @@
+import { useEffect, useState } from "react"
+import { Flex } from "../styled-system/jsx"
+import { styled } from "../styled-system/jsx"
+import PokemonCard from "./components/PokemonCard"
+import { getMultiple, getPokemon } from "./routes"
+import Favorites from "./components/Favorites"
+import { Pokemon } from "./interface"
+
+const FIRST_N: number = 4
+
 function App() {
-  return <>Hello world</>
+  const [pokemons, setPokemons] = useState<string[]>([])
+  const [favorites, setFavorites] = useState<Pokemon[]>([])
+  const [count, setCount] = useState<number>(FIRST_N)
+  const [loading, setLoading] = useState<boolean>(true)
+  const [open, setOpen] = useState<boolean>(false)
+  const [theme, setTheme] = useState<"base"|"dark">("base")
+  const [themeBox, setThemeBox] = useState<"none"|"flex">("none")
+
+  useEffect(() => {
+    if (loading === true) {
+      fetch(getMultiple(FIRST_N), {cache: "no-store"})
+      .then(res => res.json())
+      .then(res => {
+        res.results.forEach((pokemon: { name: string }) => {
+          setPokemons(pokemons => [...pokemons, pokemon.name])
+        })
+      })
+      setLoading(false)
+    } else {
+      fetch(getPokemon(count), {cache: "no-store"})
+      .then(res => res.json())
+      .then(res => {
+        setPokemons(pokemons => [...pokemons, res.name])
+      })
+      .catch(err => {
+        console.error(err)
+      })
+    }
+
+    const onScroll = () => {
+      if (document.body.scrollHeight - 200 <= window.scrollY + window.innerHeight) {
+        setCount(count => count + 1)
+      }
+    }
+      
+    window.addEventListener('scroll', onScroll, { passive: true })
+    return () => window.removeEventListener('scroll', onScroll)
+  }, [count])
+
+
+  return (
+    <Flex direction="column" fontFamily="'Poppins', sans-serif" data-theme={theme}>
+      <Favorites 
+      theme={theme}
+      isOpen={open}
+      onClose={() => setOpen(false)}
+      pokemons={favorites}
+      addFavorite={(e: Pokemon) => setFavorites(favorites => [...favorites, e])} 
+      removeFavorite={(e: string) => setFavorites(favorites => favorites.filter(fav => fav.name !== e))} 
+      ></Favorites>
```

Comment:
Kada ne pass-amo children u komponentu, moze se koristit i self closing tag
```suggestion
      />
```